### PR TITLE
Add types for container and net_id parameters in docker/api/network

### PR DESCRIPTION
**Pull request by adamtheturtle**
Some of this is akin to https://github.com/python/typeshed/blob/6246a38e8ee671bc1260f11280bb22a70bd04b31/stubs/docker/docker/api/container.pyi#L10-L16.

### CODE REVIEW

File: stubs/docker/docker/api/network.pyi [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
+
+@type_check_only
+class _HasId(TypedDict):
+    Id: str
+
+@type_check_only
+class _HasID(TypedDict):
+    ID: str
+
+_Network: TypeAlias = _HasId | _HasID | str
```
Comment (adamtheturtle) [inline stubs/docker/docker/api/network.pyi:13]: Some methods are decorated with `@check_resource('net_id')`, and I have given those methods this type.

Comment (adamtheturtle) [reply stubs/docker/docker/api/network.pyi:13]: Should we ditch this and instead say that the type is instead `docker.models.containers.Container | str`, which is perhaps more what the intention is?

**Comment by adamtheturtle** [comment]
I'm very open to specific suggestions of DRY-ing this code.

**Review by srittau — APPROVED** [review]

**Comment by srittau** [comment]
If these types (`_HasId` etc) turn up in more files, it may make sense to move them into a common location and import them.

**Comment by adamtheturtle** [comment]
They will do @srittau . I can do that when it comes up.

**Pull request merged**